[tvOS] Enable ElementFullscreen at build time

danae404 opened the pull request:
#### 3f065cd81cb4c65dbef86f15a5fcdf92f701ad56
<pre>
[tvOS] Enable ElementFullscreen at build time
<a href="https://bugs.webkit.org/show_bug.cgi?id=262824">https://bugs.webkit.org/show_bug.cgi?id=262824</a>
rdar://116604112

Reviewed by Jer Noble.

Sets ENABLE_FULLSCREEN_API to 1 on tvOS and replaces AVBackground, which is unavailable on tvOS, with a UIView in element fullscreen path.
Also removes other code that is unavailable on tvOS.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferences.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
(-[WKFullScreenViewController setPrefersHomeIndicatorAutoHidden:]):
(-[WKFullScreenViewController loadView]):
* Source/WebKit/UIProcess/ios/fullscreen/WKFullscreenStackView.h:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullscreenStackView.mm:
(SOFT_LINK_CLASS_OPTIONAL):
(-[WKFullscreenStackView init]):
(-[WKFullscreenStackView addArrangedSubviewForTV:]):
(-[WKFullscreenStackView addArrangedSubview:applyingMaterialStyle:tintEffectStyle:]):

Canonical link: <a href="https://commits.webkit.org/269172@main">https://commits.webkit.org/269172@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/09879191d5731eaaa859b6721133f85eadc8601c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21738 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/21994 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22785 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23605 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20126 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/26207 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22271 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21268 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21966 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21571 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18834 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24459 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/18754 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/19686 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25975 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/18914 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19779 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19899 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23835 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/21114 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/20378 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/17354 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/25148 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19706 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5936 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5194 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23931 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/26416 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/20299 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5768 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

jernoble commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
```diff
@@ -514,6 +516,7 @@ - (void)setLocation:(NSString *)location
 
 - (void)loadView
 {
+    WTFReportBacktrace();
```
Still needed?

danae404 replied:
Haha nope! Thanks for finding that.

rr-codes commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullscreenStackView.mm:
```diff
@@ -56,25 +61,35 @@ - (instancetype)init
+#if PLATFORM(APPLETV)
+- (void)addArrangedSubviewForTV:(UIView *)subview
+{
+    [_backgroundView.get() addSubview:subview];
```
I don't think the `.get()` here is needed

rr-codes commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullscreenStackView.mm:
```diff
@@ -56,25 +61,35 @@ - (instancetype)init
+    [self addArrangedSubview:subview];
+}
+#else
+- (void)addArrangedSubviewaddArrangedSubview:(UIView *)subview applyingMaterialStyle:(AVBackgroundViewMaterialStyle)materialStyle tintEffectStyle:(AVBackgroundViewTintEffectStyle)tintEffectStyle
```
should be `addArrangedSubview` instead of `addArrangedSubviewaddArrangedSubview`

danae404 replied:
Lol thanks! Sloppy proofread

rr-codes commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
```diff
@@ -123,13 +123,17 @@ - (CGSize)intrinsicContentSize
 
 - (void)contextMenuInteraction:(UIContextMenuInteraction *)interaction willDisplayMenuForConfiguration:(UIContextMenuConfiguration *)configuration animator:(id<UIContextMenuInteractionAnimating>)animator
 {
+#if !PLATFORM(APPLETV)
```
is this actually needed anymore now that there's the additional guard in PlatformEnableCocoa?

danae404 replied:
No, you're right

danae404 committed:
[tvOS] Enable ElementFullscreen at build time
https://bugs.webkit.org/show_bug.cgi?id=262824
rdar://116604112

Reviewed by Jer Noble.

Sets ENABLE_FULLSCREEN_API to 1 on tvOS and replaces AVBackground, which is unavailable on tvOS, with a UIView in element fullscreen path.
Also removes other code that is unavailable on tvOS.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferences.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
(-[WKFullScreenViewController setPrefersHomeIndicatorAutoHidden:]):
(-[WKFullScreenViewController loadView]):
* Source/WebKit/UIProcess/ios/fullscreen/WKFullscreenStackView.h:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullscreenStackView.mm:
(SOFT_LINK_CLASS_OPTIONAL):
(-[WKFullscreenStackView init]):
(-[WKFullscreenStackView addArrangedSubviewForTV:]):
(-[WKFullscreenStackView addArrangedSubview:applyingMaterialStyle:tintEffectStyle:]):

Canonical link: https://commits.webkit.org/269172@main
